src/drawers/rectangleDrawer.ts

```diff
@@ -5,52 +5,54 @@
 
 import * as d3 from "d3";
 
-import { Dataset } from "../core/dataset";
-
-import { Drawer } from "./drawer";
+import { AttributeToAppliedProjector } from "../core/interfaces";
+import { CanvasDrawStep } from "./canvasDrawer";
 import { AppliedDrawStep } from "./index";
+import { SVGDrawer } from "./svgDrawer";
 
-export class Rectangle extends Drawer {
-
-  constructor(dataset: Dataset) {
-    super(dataset);
-    this._svgElementName = "rect";
+export class RectangleSVGDrawer extends SVGDrawer {
+  constructor(private _rootGClassName: string = null) {
+    super("rect", "");
   }
 
-  protected _drawStepCanvas(data: any[], step: AppliedDrawStep) {
-    const context = this.canvas().node().getContext("2d");
-
-    const attrToAppliedProjector = step.attrToAppliedProjector;
-    context.save();
-
-    data.forEach((point, index) => {
-      const resolvedAttrs = Object.keys(attrToAppliedProjector).reduce((obj, attrName) => {
-        obj[attrName] = attrToAppliedProjector[attrName](point, index);
-        return obj;
-      }, {} as { [key: string]: any | number | string });
+  public draw(root: d3.Selection<SVGElement, any, any, any>, data: any[], appliedDrawSteps: AppliedDrawStep[]) {
+    if (this._rootGClassName != null) {
+      root.classed(this._rootGClassName, true);
+    }
+    super.draw(root, data, appliedDrawSteps);
+  }
+}
 
-      context.beginPath();
-      context.rect(resolvedAttrs["x"], resolvedAttrs["y"], resolvedAttrs["width"], resolvedAttrs["height"]);
-      if (resolvedAttrs["stroke-width"]) {
-        context.lineWidth = resolvedAttrs["stroke-width"];
-      }
-      if (resolvedAttrs["stroke"]) {
-        const strokeColor = d3.color(resolvedAttrs["stroke"]);
-        if (resolvedAttrs["opacity"]) {
-          strokeColor.opacity = resolvedAttrs["opacity"];
-        }
-        context.strokeStyle = strokeColor.rgb().toString();
-        context.stroke();
+export const RectangleCanvasDrawStep: CanvasDrawStep = (
+  context: CanvasRenderingContext2D,
+  data: any[],
+  attrToAppliedProjector: AttributeToAppliedProjector) => {
+  data.forEach((point, index) => {
+    const resolvedAttrs = Object.keys(attrToAppliedProjector).reduce((obj, attrName) => {
+      obj[attrName] = attrToAppliedProjector[attrName](point, index);
+      return obj;
+    }, {} as { [key: string]: any | number | string });
+
+    context.beginPath();
+    context.rect(resolvedAttrs["x"], resolvedAttrs["y"], resolvedAttrs["width"], resolvedAttrs["height"]);
```

Comment:
i believe in the scatter plot PR ben has pulled this out into a method that might be used by multiple canvas renders since many of them will want to set the styles from attrs.